### ISSUE TITLE
Email list creates but subscribers are never stored

### ISSUE THREAD

**Issue by chiefoleka** [freeCodeCamp/mail-for-good#189]
There seems to be a problem with the adding subscribers to the list.
Email list gets created (I can have like 1 million of them), but the subscribers never quite get added to the list. This happens using docker or local setup.

**Comment by karuppiah7890** [comment]
@chiefoleka Can you show a sample of you list including the header row ?

**Comment by chiefoleka** [comment]
I tried different combinations
> email
> email@gmail.com

Then I also tried
> email,first name
> email@gmail.com, firstname

Then also flipped this
> first name, email
> firstname, email@gmail.com

So, the thing is that the list comes in, it sees the headers and identifies each field irrespective of the combination I use, but it doesn't store it.

**Comment by karuppiah7890** [comment]
@chiefoleka Okay, I think I know what the issue is. I have encountered it, but I forgot to create an issue for this. 

How big is your list ? Try uploading a subset of your list with say 500-1000 emails. 

I have encountered issues in uploading when the list was big - anything more than 2000 records (only emails). The upload didn't happen properly. It's some bug.

And the output that you see for the identification (the sample), it's done on the client side and yes, looks like your list is in the right format. It just needs to have at least one column with the name "email" or even "Email" works I guess ? cc @AndrewGHC

**Comment by chiefoleka** [comment]
@karuppiah7890
Not necessarily. I tested with a single email because I felt it was the volume, but it still didn't work.
Now, it says upload completed but records 0.
The first time I set it up, I'm not sure, but I think it uploaded just fine, and I was running it in a docker container.
But before trying to use it last night, I sort of rebased my local repo people running docker. This issue started after then

**Comment by karuppiah7890** [comment]
@chiefoleka Then you could just stop and remove all containers using `docker-compose down` and try again if you don't have any important data in the DB.

@AndrewGHC I think we need to work on DB backup procedure (DB container) docs for people and provide volume based storage for DB in the upcoming versions so that it's easy to upgrade to the latest version of the application. And if for any reason we change DB schema and containers don't work in a plug and play manner, we can think on the DB upgrading procedures too later.

**Comment by chiefoleka** [comment]
@karuppiah7890
I'm not sure that would work because today, I removed all my docker containers since they are not necessarily useful yet.
But let me try and give you feedback

**Comment by chiefoleka** [comment]
@karuppiah7890 
if it helps, here is the error that comes up every time I drop and recreate a container
``` app_1       | (node:51) DeprecationWarning: Using the automatically created return value from client.query as an event emitter is deprecated and will be removed in pg@7.0. Please see the upgrade guide at https://node-postgres.com/guides/upgrading
postgres_1  | ERROR:  relation "users" does not exist at character 112
postgres_1  | STATEMENT:  SELECT "id", "googleId", "picture", "token", "email", "name", "sentEmailsCount", "createdAt", "updatedAt" FROM "users" AS "user";
```

**Comment by karuppiah7890** [comment]
That's just an error when there are no users in the DB and when the App tries to run that statement to check before logging in. I see that too in a new container. @AndrewGHC I think we could handle this error this in a better manner ?

**Comment by karuppiah7890** [comment]
@chiefoleka If you still run into issues, ping me on gitter @karuppiah7890 and I will remote login to your PC through TeamViewer and see what the issue is :)

**Comment by karuppiah7890** [comment]
@chiefoleka Any updates on the issue ?

**Comment by chiefoleka** [comment]
Same issue I have always been having. It has not Vern fixed. Everything
works, but once I get to the point where I have add list, it doesn't work.

**Comment by karuppiah7890** [comment]
"Vern" ?
And have you tried getting rid of the containers and recreating them using the commands I mentioned ?

**Comment by karuppiah7890** [comment]
@chiefoleka If you are still getting issues, let me check and verify the issue through TeamViewer. Ping me on gitter.im and we can decide on a time to check the issue

**Comment by karuppiah7890** [comment]
My Gitter username is karuppiah7890 , the same as my GitHub one

**Comment by chiefoleka** [comment]
We would do this later.

**Comment by bmorelli25** [comment]
I think I figured out what the problem is. I too was having 0 emails come through, no matter my list size (tried 0, 100, and 1000 placeholder emails).

The fix, was renaming my email column. The email column must be named exactly `email`. I had it named `Email` with a capital E. Once I lowercased everything, my subscribers uploaded for all test sizes (0, 100, and 1000).

**Comment by bmorelli25** [comment]
@chiefoleka - make sure your header is titled exactly `email`. You may have an inadvertent character, capitalization, or space that's causing the issue. I'd delete the whole header row and add a new one just to be sure.

**Comment by karuppiah7890** [comment]
@bmorelli25 Thanks for that :smile: I think I have tried with `Email` as header. Let me try again. That's probably the issue as @chiefoleka couldn't get even 1 mail to import

**Comment by deltagolf** [comment]
As @bmorelli25 has said, I solved this issue by changing the header to "email" instead of "Email".

**Comment by deltagolf** [comment]
I've been searching the code and I think changing this line would fix this:

https://github.com/freeCodeCamp/mail-for-good/blob/d1983c30c0ac6a7b9ec20e1ad3924dd506165eda/server/controllers/list/import-csv.js#L35

**Comment by pickleshy** [comment]
I too solved the issue by lowercasing "email" – would be very helpful if the helptext could be explicit about capitalization.
